### PR TITLE
review and dev notes + fixing some tests + implementations later

### PR DESCRIPTION
**Pull request by livingrockrises**
## PR-Codex overview
This PR introduces a new `solady` dependency, adds a `SmartSessionModule`, updates constants, and refactors test files.

### Detailed summary
- Added `solady` dependency
- Introduced `SmartSessionModule`
- Updated constants and modules
- Refactored test files

> The following files were skipped due to too many changes: `src/account/NexusSmartAccount.ts`, `src/modules/utils/Types.ts`, `tests/account.write.test.ts`, `src/modules/validators/SmartSessionModule.ts`, `tests/smart.sessions.write.test.ts`, `src/modules/utils/SmartSessionHelpers.ts`, `src/modules/utils/abi.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/account/NexusSmartAccount.ts [hunk 1]
```diff
@@ -64,7 +65,7 @@ import {
   convertSigner
 } from "./index.js"
 import {
-  GENERIC_FALLBACK_SELECTOR,
+  GENERIC_FALLBACK_SELECTOR_SELECTOR,
```
Comment (livingrockrises) [inline src/account/NexusSmartAccount.ts:68]: should not be importing this in nexus. it was just a selector for a mock fallback.
@VGabriel45

File: src/account/NexusSmartAccount.ts [hunk 2]
```diff
@@ -2073,16 +2078,18 @@ export class NexusSmartAccount extends BaseSmartContractAccount {
...
-        : module.type === "executor"
+        : moduleType === "executor"
           ? 2n
-          : module.type === "fallback"
+          : moduleType === "fallback"
             ? 3n
-            : 4n
+            : moduleType === "hook"
+              ? 4n
+              : 0n // review
```
Comment (livingrockrises) [inline src/account/NexusSmartAccount.ts:2092]: default should be 0?
there may be more module types in the future.
in consensus with ERC7579

File: tests/account.write.test.ts [hunk 3]
```diff
@@ -20,9 +24,15 @@ import {
   topUp
 } from "./src/testUtils"
 import type { MasterClient, NetworkConfig } from "./src/testUtils"
+import { GENERIC_FALLBACK_SELECTOR_SELECTOR } from "../src/bundler/utils/Constants"
 
 const NETWORK_TYPE: TestFileNetworkType = "FILE_LOCALHOST"
 
+// todo
+// remove hard code and fetch from deployments
```
Comment (livingrockrises) [inline tests/account.write.test.ts:32]: @joepegler was asking for this part. probably need help here and update after our all deployments are sourced in a chosen way.

File: tests/account.write.test.ts [hunk 4]
```diff
@@ -118,110 +128,103 @@ describe("account.write", () => {
...
+      const isInstalled = await smartAccount.isModuleInstalled({
+        moduleAddress: MOCK_HOOK,
+        type: "hook"
+      })
+
+      expect(success).toBe(true)
+      expect(isInstalled).toBeFalsy()
+    }, 60000)
+
+    test("install a fallback handler Hook module", async () => {
```
Comment (livingrockrises) [inline tests/account.write.test.ts:190]: these can't run until installModule and uinstallModule is properly written in Nexus. along with accepted types reviews

File: tests/account.write.test.ts [hunk 5]
```diff
@@ -118,110 +128,103 @@ describe("account.write", () => {
...
+        moduleAddress: MOCK_FALLBACK_HANDLER,
+        data: encodeAbiParameters(
+          [{ type: 'bytes4' }],
+          [ GENERIC_FALLBACK_SELECTOR_SELECTOR as Hex ]
+        )
+      })
+      console.log(isInstalledBefore, "is installed before")
+    }, 60000)
+
+    // test("uninstall handler module", async () => {
```
Comment (livingrockrises) [inline tests/account.write.test.ts:205]: these can't run until installModule and uinstallModule is properly written in Nexus. along with accepted types reviews

**Comment by livingrockrises** [comment]
we need to get smart session deployment for our tests and real chain wide constant. cc @joepegler

File: tests/account.write.test.ts (hunk 4 above)
Comment (VGabriel45) [reply tests/account.write.test.ts:190]: I dont get this one @livingrockrises  wdym when you say "written in Nexus" in Nexus SDK or Nexus SC ?

Comment (livingrockrises) [reply tests/account.write.test.ts:191]: I meant in the SA file

**Pull request closed without merging**